From 8235d904237b457ed57dc151cb4f40fb178749ac Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Tue, 3 Feb 2026 05:56:14 -0500
Subject: [PATCH 7/7] AOSCOS: Adjust mozconfig

- Set prefix to /usr.
- Disable updater.
- Update dependency settings.
- Disable boostrapping.

Ref: aosc-os-abbs/app-web/firefox.
---
 configs/common/mozconfig | 13 ++------
 configs/linux/mozconfig  | 69 +++++++++++++++++++++-------------------
 2 files changed, 38 insertions(+), 44 deletions(-)

diff --git a/configs/common/mozconfig b/configs/common/mozconfig
index 673f54d70..6648bd1c6 100644
--- a/configs/common/mozconfig
+++ b/configs/common/mozconfig
@@ -6,9 +6,6 @@
 ac_add_options --with-app-name=${binName}
 ac_add_options --with-app-basename=Zen
 
-# Localization (Must be an absolute path)
-ac_add_options --with-l10n-base="${topsrcdir}/browser/locales"
-
 export MOZ_APP_BASENAME=Zen
 export MOZ_BRANDING_DIRECTORY=${brandingDir}
 export MOZ_OFFICIAL_BRANDING_DIRECTORY=${brandingDir}
@@ -53,8 +50,8 @@ fi
 
 if test "$ZEN_RELEASE"; then
 
-  ac_add_options --enable-clang-plugin
-  ac_add_options --enable-bootstrap=-sccache
+  # ac_add_options --enable-clang-plugin
+  # ac_add_options --enable-bootstrap=-sccache
 
   ac_add_options --enable-optimize
 
@@ -65,7 +62,6 @@ if test "$ZEN_RELEASE"; then
   ac_add_options --disable-tests
 
   ac_add_options --enable-rust-simd
-  ac_add_options --enable-wasm-simd
 
   ac_add_options --disable-geckodriver
   ac_add_options --disable-rust-tests
@@ -92,11 +88,6 @@ if test "$ZEN_RELEASE"; then
   export MOZ_PACKAGE_JSSHELL=1
 
   ac_add_options --disable-crashreporter
-
-  # Experimental flag, enabled only on nightly for Firefox.
-  # Should bring in some nice performance improvements,
-  # but may cause stability issues.
-  ac_add_options --enable-replace-malloc
 fi
 
 ac_add_options --enable-jxl
diff --git a/configs/linux/mozconfig b/configs/linux/mozconfig
index 9816e7580..2634d6e4a 100644
--- a/configs/linux/mozconfig
+++ b/configs/linux/mozconfig
@@ -11,36 +11,39 @@ else
     export CXX=clang++
 fi
 
-if test "$ZEN_RELEASE"; then
-    if test "$SURFER_COMPAT" = "x86_64"; then
-        ac_add_options --target=x86_64-pc-linux
-        ac_add_options --enable-eme=widevine
-
-        # Enable Profile Guided Optimization
-        if ! test "$ZEN_GA_DISABLE_PGO"; then
-            if test "$ZEN_GA_GENERATE_PROFILE"; then
-                mk_add_options "export MOZ_AUTOMATION_PACKAGE_GENERATED_SOURCES=0"
-                ac_add_options --enable-profile-generate=cross
-            else
-                ac_add_options --enable-profile-use=cross
-                ac_add_options --with-pgo-profile-path="$(echo ~)/artifact/merged.profdata"
-                ac_add_options --with-pgo-jarlog="$(echo ~)/artifact/en-US.log"
-            fi
-        fi
-    elif test "$SURFER_COMPAT" = "aarch64"; then
-        ac_add_options --target=aarch64-linux-gnu
-
-        # override LTO settings
-        export MOZ_LTO=cross,thin
-        ac_add_options --enable-lto=cross,thin
-    fi
-
-    # Disable DMD and ELF hacks, enable linker lld
-    ac_add_options --enable-linker=lld
-    ac_add_options --disable-elf-hack
-
-    # Stripping options for release builds
-    ac_add_options --enable-install-strip
-    ac_add_options --enable-strip
-    export STRIP_FLAGS="--strip-debug --strip-unneeded"
-fi
+# Disable DMD and necko-wifi
+ac_add_options --disable-dmd
+ac_add_options --disable-necko-wifi
+
+# Common options
+ac_add_options --enable-alsa
+ac_add_options --enable-pulseaudio
+
+# Basic path definitions.
+ac_add_options --prefix=/usr
+ac_add_options --libdir=/usr/lib
+
+# Toolchain settings.
+ac_add_options --disable-strip
+ac_add_options --disable-install-strip
+ac_add_options --enable-hardening
+
+# Optimize
+ac_add_options --enable-optimize="-O2"
+
+# Dependencies.
+ac_add_options --disable-bootstrap
+ac_add_options --enable-system-ffi
+ac_add_options --without-system-jpeg
+ac_add_options --with-system-libevent
+ac_add_options --with-system-libvpx
+ac_add_options --with-system-nspr
+ac_add_options --with-system-nss
+ac_add_options --with-system-zlib
+ac_add_options --without-system-icu
+
+# Features.
+ac_add_options --enable-chrome-format=omni
+ac_add_options --enable-js-shell
+ac_add_options --disable-tests
+ac_add_options --disable-updater
-- 
2.52.0

